### PR TITLE
RD-10004 improve-error-propagation-for-lsp-in-the-compiler-interfaces

### PR DESCRIPTION
**Pull request by Alexzerntev**
Updated Errors class, added:
- code as a type of error message
- tags (deprecated etc..)
- severity 

These information is corespondent to what monaco editor expects

### CODE REVIEW

File: client/src/main/scala/raw/client/api/Errors.scala [hunk 1]
```diff
@@ -18,8 +18,17 @@ import raw.utils.RawException
...
  * @param positions The positions where the error occurred.
+ * @param severity The severity of the error. 1 = Hint, 2 = Info, 4 = Warning, 8 = Error
+ * @param code An optional error code.
+ * @param tags Indication for the error Unnecessary = 1, Deprecated = 2
  */
-final case class ErrorMessage(message: String, positions: List[ErrorRange])
+final case class ErrorMessage(
+    message: String,
+    positions: List[ErrorRange],
+    severity: Int = 8,
```
Comment (miguelbranco80) [inline client/src/main/scala/raw/client/api/Errors.scala:28]: Suggest you strict the API. By making it a number, I guarantee one of us will add an 'id' to it by accident ;) Make it an scala enum and use that as the API. Can use the underlying numbers in the enum definition.

File: client/src/main/scala/raw/client/api/Errors.scala [hunk 2]
```diff
@@ -18,8 +18,17 @@ import raw.utils.RawException
...
+    code: Option[String] = None,
+    tags: List[Integer] = List.empty
```
Comment (miguelbranco80) [inline client/src/main/scala/raw/client/api/Errors.scala:30]: BTW, this is Integer, not Int. In any case, same comment as before: use a scala enum.

**Review by bgaidioz — APPROVED** [review]

**Comment by Alexzerntev** [comment]
To be closed, the final implementation will be provided here: https://github.com/raw-labs/snapi/pull/359

**Review by bgaidioz — APPROVED** [review]

**Pull request closed without merging**